src/core/haiku/SDL_BeApp.cc

```diff
@@ -51,7 +52,7 @@ const char *signature = "application/x-SDL-executable";
 
 static int StartBeApp(void *unused)
 {
-    BApplication *App;
+    std::unique_ptr<BApplication> App;
 
     // dig resources for correct signature
```

Comment:
100% untested.
Does haiku emit warnings when building with `-Wunused`?

```suggestion
    (void)unused;
    // dig resources for correct signature
```

Reply:
this or an unused attribute. I had no issue at compile with my Haiku box.